Updated/fixed 3 exclusive driver related natives

chipjpg opened the pull request.

Disquse commented on VEHICLE/SetVehicleActiveSomething.md:
````diff
@@ -0,0 +1,22 @@
+
+```c
+// 0x41062318F23ED854
+void SET_VEHICLE_ACTIVE_SOMETHING(Vehicle vehicle, cs_type(Ped) BOOL toggle);
````
If this name is not confirmed, you need to add `_` prefix to the name.

gottfriedleibniz commented:
May need to double check ABI compatibility on `0xB09D25E77C33EB3F` (cs_type int?). Additionally, `0x41062318F23ED854` has an `SET_VEHICLE_EXCLUSIVE_DRIVER` alias which conflicts with `0xB5C51B5502E85E83`.

In addition, this native has been incorrect for so long, that changing it at this point will likely break scripts that have compensated for this mess. Just something to consider.

chipjpg commented:
> Additionally, `0x41062318F23ED854` has an `SET_VEHICLE_EXCLUSIVE_DRIVER` alias which conflicts with `0xB5C51B5502E85E83`

Yeah I added that alias for documentation of prior name purposes but I see the issue now. I've removed that alias now.

> May need to double check ABI compatibility on `0xB09D25E77C33EB3F`

How would I do that? The parameter type was previously set as an integer but after looking at the disassembly, it's clearly an int*.

I've changed the file to include cs_type int now though.

stannum-cfx commented on VEHICLE/IsPedExclusiveDriverOfVehicle.md:
````diff
@@ -1,17 +1,18 @@
 ```c
-// 0xB09D25E77C33EB3F
-BOOL _IS_PED_EXCLUSIVE_DRIVER_OF_VEHICLE(Ped ped, Vehicle vehicle, int outIndex);
+// 0xB09D25E77C33EB3F 0xFF62D324
````
I'm not sure adding joaat hashes is supported, this might break existing resources. 

Where did you get this hash from?

chipjpg replied:
The joaat hash was taken from [VEHICLE/SetVehicleExclusiveDriver.md](https://github.com/citizenfx/natives/pull/625/files/f18980e435cce3f26e18a51f9f792c14e0aad3bd#diff-b48a74e0d1a1f86a6838ac21ddf2e4ed0835746e2b1735a7d619b440520d6eae).

As you can see, the previous joaat hashes didn't match. 

If you hash "IS_PED_EXCLUSIVE_DRIVER_OF_VEHICLE" you will get 0xFF62D324.

stannum-cfx commented on VEHICLE/SetVehicleExclusiveDriver.md:
````diff
@@ -1,25 +1,18 @@
 ```c
-// 0x41062318F23ED854 0xFF62D324
-void SET_VEHICLE_EXCLUSIVE_DRIVER(Vehicle vehicle, cs_type(Ped) BOOL toggle);
+// 0xB5C51B5502E85E83 0xAA8BD440
````
Again, not sure if changing hashes is properly supported.
Maybe @blattersturm could verify this?

stannum-cfx reviewed (commented):
These changes are quite confusing. 

Maybe just changing the descriptions but keeping the original (incorrect) names and hashes is safer, at least for compatibility with existing scripts.

AvarianKnight commented:
Given the fact that theres no nice way to add this without breaking compatibility with existing resources I'm going to close this.
